Character.cpp

```diff
@@ -98,20 +97,37 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        if ( initialArmorLevel && initialHitPoints && initialAttackDamage ) 
+        {
+            initialHitPoints.reset( new int( *initialHitPoints ) );
+            initialArmorLevel.reset( new int( *initialArmorLevel) );
+            initialAttackDamage.reset( new int( *initialAttackDamage ) );
```

Comment:
Why are you resetting?  `initialHitPoints` holds the "default hit points" when your character is created.   `hitPoints` is the CURRENT stat.  
```
a) your stats are restored to their initial value
```
this means to set `hitPoints` back to the initial value at time of construction, which is stored in `initialHitPoints`. 
```
val = *initialVal;
```
where `val` is the stat, and `initialVal` is the unique_ptr member holding the default value at time of construction